src/gestures/__tests__/pan.test.tsx

```diff
@@ -1,45 +1,46 @@
 import * as React from "react"
 import { motion } from "../../"
 import { render } from "@testing-library/react"
-import { MockDrag, asyncDrag } from "../../behaviours/__tests__/utils"
-import sync from "framesync"
+import {
+    MockDrag,
+    drag,
+    frame,
+    deferred,
+} from "../../behaviours/__tests__/utils"
 
 describe("pan", () => {
     test("pan handlers aren't frozen at pan session start", async () => {
-        const promise = new Promise(async resolve => {
-            let count = 0
-            const Component = () => {
-                const [increment, setIncrement] = React.useState(1)
-                return (
-                    <MockDrag>
-                        <motion.div
-                            onPanStart={() => {
-                                count += increment
-                                setIncrement(2)
-                            }}
-                            onPan={() => (count += increment)}
-                            onPanEnd={() => {
-                                count += increment
-                                resolve(count)
-                            }}
-                        />
-                    </MockDrag>
-                )
-            }
+        let count = 0
+        const onPanEnd = deferred()
+        const Component = () => {
+            const [increment, setIncrement] = React.useState(1)
+            return (
+                <MockDrag>
+                    <motion.div
+                        onPanStart={() => {
+                            count += increment
+                            setIncrement(2)
+                        }}
+                        onPan={() => (count += increment)}
+                        onPanEnd={() => {
+                            count += increment
+                            onPanEnd.resolve()
+                        }}
+                    />
+                </MockDrag>
+            )
+        }
 
-            const { container, rerender } = render(<Component />)
-            rerender(<Component />)
+        const { container, rerender } = render(<Component />)
+        rerender(<Component />)
 
-            const pointer = await asyncDrag(container.firstChild).to(100, 100)
+        const pointer = await drag(container.firstChild).to(100, 100)
+        await frame.postRender()
```

Comment:
Ideally this shouldn't be needed. I've only preserved explicit calls to `sync.postRender` in those tests concerned about calling latest version of handlers, because I had to match expected counter. I believe those 2 tests could be rewritten somehow in such a way that we wouldn't have to explicitly call this API.